openapi.yaml

```diff
@@ -0,0 +1,730 @@
+        Create a new Service with the attributes provided in the request payload.
+      operationId: createService
+      requestBody:
+        description: |
+          A service body payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+        required: true
+      responses:
+        '201':
+          description: Service created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+    get:
+      tags:
+        - services
+      summary: Retrieve all services
+      description: Retrieve all services owned by the calling user
+      operationId: getServices
+      parameters: 
+        - name: limit
+          in: query
+          description: Number of services per page
+          required: true
+          schema:
+            type: number
+        - name: offset
+          in: query
+          description: Offset services number
+          required: false
+          schema:
+            type: number
+      responses:
+        '200':
+          description: Services fetched successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ServicePagination'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+  /services/{serviceId}:
+    get:
+      tags:
+        - services
+      summary: Retrieve service
+      description: Retrieve a service by `serviceId`
+      operationId: getService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Service fetched successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error
+      security:
+        - api_key: []
+    put:
+      tags:
+        - services
+      summary: Update service
+      description: |
+        Update an existing service by by `serviceId`.
```

Comment:
```suggestion
      description: Update an existing service by its id.
```

Reply:
i will remove the "|", but for the description I prefer the current version